lib/updateDataDisplay.ml

```diff
@@ -320,7 +318,7 @@ let print_foreach conf print_ast _eval_expr =
     in
```

Comment:
in `loop` we can use the match to directly see if we are on the last element and should do nothing, so it is more clear and we can remove the test `if List.length l > 0 ` : 

```ocaml
    let rec loop first cnt = function
      | [] -> ()
      | [ _s ] -> () (* dont do last element *)
      | s :: l ->
            let tail =
              if String.length s > 0 then List.nth (string_to_list s) 0
              else ""
            in
            let env =
              ("substr", Vstring s) :: ("tail", Vstring tail)
              :: ("first", Vbool first) :: ("max", Vint max)
              :: ("cnt", Vint cnt) :: env
            in
            List.iter (print_ast env xx) al;
            loop false (cnt + 1) l
    in
```

Reply:
ok